crates/biome_js_analyze/src/aria_analyzers/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,126 @@
+    /// ```
+    ///
+    /// ```jsx
+    /// <input type="text" onClick={() => {}} />;
+    /// ```
+    ///
+    pub(crate) NoStaticElementInteractions {
+        version: "next",
+        name: "noStaticElementInteractions",
+        recommended: false,
+    }
+}
+
+const INTERACTIVE_HANDLER_LIST: [&str; 6] = [
+    "onClick",
+    "onMouseDown",
+    "onMouseUp",
+    "onKeyDown",
+    "onKeyPress",
+    "onKeyUp",
+];
+
+impl Rule for NoStaticElementInteractions {
+    type Query = Aria<AnyJsxElement>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let aria_roles = ctx.aria_roles();
+        let attributes = ctx.extract_attributes(&node.attributes());
+        let is_not_interactive_element = is_not_interactive_element(node, aria_roles, attributes)?;
+        let has_interactive_handler = has_interactive_handler(node)?;
+        let has_no_interactive_role = !has_interactive_role(node, aria_roles);
+
+        (is_not_interactive_element && has_interactive_handler && has_no_interactive_role)
+            .then_some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Static elements should not be interactive."
+                },
+            )
+            .note(markup! {
+                "To add interactivity such as a mouse or key event listener to a static element, give the element an appropriate role value"
+            }),
+        )
+    }
+}
+
+fn is_not_interactive_element(
+    node: &AnyJsxElement,
+    aria_roles: &AriaRoles,
+    attributes: Option<HashMap<String, Vec<String>, BuildHasherDefault<FxHasher>>>,
+) -> Option<bool> {
+    let maybe_element_name = node.name().ok()?.as_jsx_name()?.value_token().ok()?;
+    let element_name: &str = maybe_element_name.text_trimmed();
+    let is_not_interactive = aria_roles.is_not_interactive_element(element_name, attributes);
+
+    is_not_interactive.then_some(true)
+}
+
+fn has_interactive_handler(node: &AnyJsxElement) -> Option<bool> {
+    let has_interactive_handler = INTERACTIVE_HANDLER_LIST.iter().find_map(|&handler_name| {
+        Some(
+            !node
+                .find_attribute_by_name(handler_name)?
+                .is_value_null_or_undefined(),
+        )
+    })?;
+
+    has_interactive_handler.then_some(true)
+}
+
+fn has_interactive_role(node: &AnyJsxElement, aria_roles: &AriaRoles) -> bool {
+    match node.find_attribute_by_name("role") {
+        Some(attribute) => match attribute.as_static_value() {
+            Some(value) => aria_roles.is_role_interactive(value.text()),
+            None => false,
+        },
+        None => false,
+    }
+}
```

Comment:
Is there a cleaner way to avoid the nested matches?

Reply:
You can use `let else`:

```ts
let Some(value) = node.find_attribute_by_name("role") else { return false; };
aria_roles.is_role_interactive(value.text())
```

or `if let`


```ts
if let Some(value) = node.find_attribute_by_name("role") {
    aria_roles.is_role_interactive(value.text())
} else {
    false
}
```

or `map`:

```rs
node.find_attribute_by_name("role").map(|aria_roles| aria_roles.is_role_interactive(value.text())).unwrap_or(false)
```

Reply:
In these cases, it's acceptable to return `Option<bool>` instead of `bool`. This will make the code way cleaner; then the caller will do 

```rs

if has_interactive_role()? == true {}
```